### PR TITLE
test: Add tests for duplicate language property key bug

### PR DESCRIPTION
**Pull request by dbasunag**
# Pull Request

## Related Issues

- Fixes: 
- JIRA: 

## Please review and indicate how it has been tested

- [ ] Locally
- [ ] Jenkins

## Additional Requirements

- [ ] If this PR introduces a new test image, did you create a PR to mirror it in disconnected environment?
- [ ] If this PR introduces new marker(s)/adds a new component, was relevant ticket created to update relevant Jenkins job?

## Summary by CodeRabbit

* **Tests**
  * Added health checks that scan catalog logs for duplicate-key errors.
  * Added end-to-end database-vs-API consistency tests ensuring model language properties match across sources.
  * Added regression tests validating multilingual model property handling.
  * Introduced a shared fixture for stable access to the catalog instance in tests.
  * Added helper utilities to parse and compare language properties between the database and catalog API.

### CODE REVIEW

File: tests/model_registry/model_catalog/conftest.py [hunk 1]
```diff
@@ -42,6 +43,14 @@
 LOGGER = structlog.get_logger(name=__name__)
 
 
+@pytest.fixture(scope="class")
+def model_catalog_pod(admin_client: DynamicClient, model_registry_namespace: str) -> Pod:
+    """Get the first model catalog pod in the model registry namespace."""
+    pods = get_model_catalog_pod(client=admin_client, model_registry_namespace=model_registry_namespace)
+    assert pods, "No model catalog pods found"
+    return pods[0]
```
Comment (lugi0) [inline tests/model_registry/model_catalog/conftest.py:51]: We already have `get_model_catalog_pod`

File: tests/model_registry/model_catalog/db_check/test_model_catalog_db_validation.py [hunk 2]
```diff
@@ -197,3 +213,71 @@ def test_non_catalog_network_policy_not_recreated(
...
+                continue
+            parts = line.split("|")
+            if len(parts) == 2:
+                model_name = parts[0].strip()
+                raw_value = parts[1].strip()
+                try:
+                    langs = json.loads(raw_value)
+                    db_languages.setdefault(model_name, set()).update(langs)
+                except json.JSONDecodeError:
+                    db_languages.setdefault(model_name, set()).add(raw_value)
```
Comment (lugi0) [inline tests/model_registry/model_catalog/db_check/test_model_catalog_db_validation.py]: can this be a helper function?

File: tests/model_registry/model_catalog/db_check/test_model_catalog_db_validation.py [hunk 3]
```diff
@@ -197,3 +213,71 @@ def test_non_catalog_network_policy_not_recreated(
...
+                headers=model_registry_rest_headers,
+            )
+            items = api_model.get("items", [])
+            if not items:
+                mismatches.append(f"{model_name}: not found in API")
+                continue
+
+            api_langs = set(items[0].get("language", []))
+            if db_langs != api_langs:
+                mismatches.append(f"{model_name}: DB={sorted(db_langs)}, API={sorted(api_langs)}")
```
Comment (lugi0) [inline tests/model_registry/model_catalog/db_check/test_model_catalog_db_validation.py]: can this be a helper function?

File: tests/model_registry/model_catalog/db_check/test_model_catalog_db_validation.py [hunk 4]
```diff
@@ -197,3 +213,71 @@ def test_non_catalog_network_policy_not_recreated(
...
+
+LANGUAGE_PROPERTIES_DB_QUERY = """
+SELECT c.name AS model_name, cp.string_value AS language
+FROM "Context" c
+JOIN "ContextProperty" cp ON cp.context_id = c.id
+WHERE c.type_id = (SELECT id FROM "Type" WHERE name = 'kf.CatalogModel')
+AND cp.name = 'language'
+AND cp.is_custom_property = false
+ORDER BY c.name, cp.string_value;
+"""
```
Comment (lugi0) [inline tests/model_registry/model_catalog/db_check/test_model_catalog_db_validation.py]: might be worth moving into constant

**Review by lugi0 — COMMENTED** [review]
A few comments

File: tests/model_registry/model_catalog/db_check/test_model_catalog_db_validation.py [hunk 5]
```diff
@@ -57,6 +60,19 @@ def test_model_catalog_pod_ready_after_secret_recreation(
...
+    def test_no_duplicate_key_errors_in_catalog_logs(self, model_catalog_pod):
+        """Given a model catalog pod with default data loaded
+        When checking the catalog container logs
+        Then no 'duplicated key not allowed' errors should be present
+        """
+        catalog_log = model_catalog_pod.log(container=CATALOG_CONTAINER)
+        duplicate_errors = [line for line in catalog_log.splitlines() if "duplicated key not allowed" in line]
+        assert not duplicate_errors, (
+            f"Found {len(duplicate_errors)} duplicate key errors in catalog logs:\n" + "\n".join(duplicate_errors)
+        )
```
Comment (fege) [inline tests/model_registry/model_catalog/db_check/test_model_catalog_db_validation.py]: do you really need to count how many? the error simple should not be present, you can use `assert "duplicated key not allowed" not in catalog_log`

File: tests/model_registry/model_catalog/conftest.py (hunk 1 above)
Comment (dbasunag) [reply tests/model_registry/model_catalog/conftest.py:51]: > Yes. It creates a fixture utilizing that utility function.

File: tests/model_registry/model_catalog/db_check/test_model_catalog_db_validation.py (hunk 2 above)
Comment (dbasunag) [reply tests/model_registry/model_catalog/db_check/test_model_catalog_db_validation.py]: Only used in one place, hence I kept it like this. But if you prefer a utility, I can definitely create one.

File: tests/model_registry/model_catalog/db_check/test_model_catalog_db_validation.py (hunk 3 above)
Comment (dbasunag) [reply tests/model_registry/model_catalog/db_check/test_model_catalog_db_validation.py]: Same answer, it is only used in one place.

File: tests/model_registry/model_catalog/db_check/test_model_catalog_db_validation.py (hunk 4 above)
Comment (dbasunag) [reply tests/model_registry/model_catalog/db_check/test_model_catalog_db_validation.py]: Sure.

**Review by fege — APPROVED** [review]
/lgtm

**Review by lugi0 — APPROVED** [review]
